lib/stable_email.py

```diff
@@ -0,0 +1,178 @@
+    raw_msg = commit.tree["m"].data_stream.read()
+    m = email.message_from_bytes(raw_msg, policy=email.policy.EmailPolicy(utf8=True))
+    return m
+
+
+def is_greg_request(commit):
+    if commit.author.email == "gregkh@linuxfoundation.org":
+        m = commit_to_email_message(commit)
+        if "X-KernelTest-Branch" in m and "in-reply-to" not in m:
+            return m
+    return False
+
+
+def is_beyond_time_search(commit, limit):
+    if commit.committed_datetime < limit:
+        return True
+    return False
+
+
+def get_version(m):
+    sub = m["subject"]
+    if not sub.endswith("-stable review"):
+        return False
+    if not sub.startswith("["):
+        return False
+    sub1 = sub.replace("-stable review", "")
+    return sub1[sub1.rfind(" ") + 1 :]
+
+
+def get_review_requests(dt_limit):
+    print(
+        "* Looking for review requests after %s..."
+        % dt_limit.strftime("%Y-%m-%d %H:%M UTC")
+    )
+    fg = {}
+    x = 0
+    repo = git.Repo("/data/lore-stable")
+    commits = repo.iter_commits("HEAD")
+
+    for commit in commits:
+        # Limit search
+        if is_beyond_time_search(commit, dt_limit):
+            print("Done. Found %d review requests in %d messages." % (len(fg), x))
+            break
+
+        # Look for Greg's stable RC review requests
+        msg = is_greg_request(commit)
+        if msg:
+            print("Found: %s" % commit.summary)
+            fg[msg["message-id"]] = {"request": commit}
+        x += 1
+
+    return fg
+
+
+def get_review_replies(oldest, fg):
+    print("* Looking for replies...")
+    x = 0
+    repo = git.Repo("/data/lore-stable")
+    commits = repo.iter_commits("HEAD")
+
+    for commit in commits:
+        # Limit search
+        if is_beyond_time_search(commit, oldest):
+            print("Done. (Looked at %d messages.)" % x)
+            break
+
+        msg = commit_to_email_message(commit)
+
+        if "in-reply-to" in msg and msg["in-reply-to"] in fg:
+            inrt = msg["in-reply-to"]
+            efrom = msg["from"]
+            # if 'gregkh' not in efrom:
+            if "linaro.org" in efrom:
+                print("%d: %s" % (x, efrom))
+                if "replies" in fg[inrt]:
+                    fg[inrt]["replies"].append(commit)
+                else:
+                    fg[inrt]["replies"] = [commit]
+
+        x += 1
+
+    # print(str(from_greg))
+    return fg
+
+
+class Review(object):
+    request = None
+    reply = None
+    elapsed_time = None
+
+    def __init__(self, req, reply):
+        self.request = req
+        self.reply = reply
+
+    def calc_elapsed_time(self):
+        request_time = self.request.committed_datetime
+        if self.reply:
+            reply_time = self.reply.committed_datetime
+
+            # This is a datetime.timedelta
+            self.elapsed_time = reply_time - request_time
+
+    def get_elapsed_time(self):
+        self.calc_elapsed_time()
+
+        et = self.elapsed_time
+        if et:
```

Comment:
This would be improved by handing the exceptional case in the conditional and then exiting the conditional. In general, this reduces the nesting of conditions and makes for cleaner code imho. For example:

```
if not et:
    return None
<stuff that used to be inside a conditional block>
```